### ISSUE TITLE
How to use auth basic?

### ISSUE THREAD

**Issue by kvityasya** [tiaguinho/gosoap#5]

**Comment by tiaguinho** [comment]
Can you provide an example?

**Comment by kvityasya** [comment]
I did not find where to send the headers

**Comment by kvityasya** [comment]
this simple not work
```go
func basicAuth(username, password string) string {
	auth := username + ":" + password
	return base64.StdEncoding.EncodeToString([]byte(auth))
}

func main() {
	soap, err := gosoap.SoapClient("https://site.ru?wsdl")
	if err != nil {
		fmt.Errorf("error not expected: %s", err)
	}
	soap.HeaderParams = gosoap.Params{
		"Authorization": "Basic " + basicAuth("username","password"),
	}
	params := gosoap.Params{}
	err = soap.Call("GetBrands", params)
	fmt.Print(err)
	if err != nil {
		fmt.Errorf("error in soap call: %s", err)
	}
}
```

**Comment by Fale** [comment]
This gets fixed by either #4 and #8
